lib/coney/rabbit_connection.ex

```diff
@@ -3,29 +3,36 @@ defmodule Coney.RabbitConnection do
 
   require Logger
 
-  def open(settings = %{url: url, timeout: timeout}) do
-    case connect(url) do
+  @timeout 1000
+
+  def open(settings) do
+    case connect(settings) do
       {:ok, conn} ->
-        Logger.debug("#{__MODULE__} (#{inspect(self())}) connected to #{url}")
+        Logger.debug("#{__MODULE__} (#{inspect(self())}) connected to server")
 
         Process.monitor(conn.pid)
         conn
 
       {:error, error} ->
         Logger.error(
-          "#{__MODULE__} (#{inspect(self())}) connection to #{url} refused: #{inspect(error)}"
+          "#{__MODULE__} (#{inspect(self())}) connection to server refused: #{inspect(error)}"
         )
 
-        :timer.sleep(timeout)
+        :timer.sleep(@timeout)
         open(settings)
     end
   end
 
-  defp connect(url) do
+  defp connect(%{url: url}) do
     url
     |> choose_server()
     |> Connection.open()
   end
+  defp connect(settings = %{}) do
```

Comment:
In that case it won't be possible to specify connection settings for multiple hosts. Currently you can specify list of rabbitmq connection urls and coney will connect to random instance. If connection lost, it will connect to another rabbitmq server.

Reply:
We have the same requirements at work: that is to reconnect to another node if one of them fails. But we do that with a load balancer.

I've asked, on RabbitMQ workspace on Slack, why so many clients don't offer that feature, and the answer is it's because they solve it at the infrastructure level, not at the client one. 

So I don't think that feature is really useful, in the end.

Reply:
Good point, that's how it should work, but I had situations when it worked without load balancer and client should handle it by itself, that's why I made this feature.